RUFAS/biophysical/animal/reproduction/reproduction.py

```diff
@@ -1,2 +1,2768 @@
+    heifer_reproduction_sub_program : HEIFER_REPRODUCTION_SUB_PROTOCOLS, optional
+        The sub-program for heifer reproduction, by default None.
+    cow_reproduction_program : CowReproductionProtocol, optional
+        The reproduction program for cows, by default None.
+    ai_day : int, optional
+        The day of artificial insemination, by default 0.
+    estrus_day : int, optional
+        The day estrus is expected, by default 0.
+    abortion_day : int, optional
+        The day of abortion, by default 0.
+    breeding_to_preg_time : int, optional
+        Time taken from breeding to pregnancy, by default 0.
+    conception_rate : float, optional
+        Conception rate of the animal, by default 0.0.
+    cow_TAI_conception_rate : float, optional
+        Timed artificial insemination (TAI) conception rate for cows, by default 0.0.
+    num_conception_rate_decreases : int, optional
+        Number of times the conception rate decreases, by default 0.
+    hormone_schedule : dict[int, dict[str, Any]], optional
+        The hormone schedule for the reproduction protocol, by default None.
+    gestation_length : int, optional
+        Length of the gestation period, by default 0.
+    conceptus_weight : float, optional
+        Weight of the conceptus, by default 0.0.
+    calf_birth_weight : float, optional
+        Birth weight of the calf, by default 0.0.
+    calves : int, optional
+        Number of calves, by default 0.
+    calving_interval : int, optional
+        Interval between calvings, by default 0.
+    calving_interval_history : list[int], optional
+        History of calving intervals, by default None.
+    body_weight_at_calving : float, optional
+        Body weight of the animal at calving, by default 0.0.
+    do_not_breed : bool, optional
+        Flag indicating if the animal should not breed, by default False.
+    """
+
+    def __init__(
+            self,
+            heifer_reproduction_program: HeiferReproductionProtocol = None,
+            heifer_reproduction_sub_program: HEIFER_REPRODUCTION_SUB_PROTOCOLS = None,
+            cow_reproduction_program: CowReproductionProtocol = None,
+            cow_presynch_program: CowPreSynchSubProtocol = None,
+            cow_ovsynch_program: CowTAISubProtocol = None,
+            cow_resynch_program: CowReSynchSubProtocol = None,
+            ai_day: int = 0,
+            estrus_day: int = 0,
+            abortion_day: int = 0,
+            breeding_to_preg_time: int = 0,
+            conception_rate: float = 0.0,
+            cow_TAI_conception_rate: float = 0.0,
+            num_conception_rate_decreases: int = 0,
+            hormone_schedule: dict[int, dict[str, Any]] = None,
+            gestation_length: int = 0,
+            conceptus_weight: float = 0.0,
+            calf_birth_weight: float = 0.0,
+            calves: int = 0,
+            calving_interval: int = 0,
+            calving_interval_history: list[int] = None,
```

Comment:
```suggestion
            calving_interval_history: list[int] | None = None,
```

Could then initialize this:
self.calving_interval_history = calving_interval_history or []